src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -77,11 +77,31 @@ public void visitToken(DetailAST ast) {
             final DetailAST lastGroupAST =
                 switchAST.getLastChild().getPreviousSibling();
 
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
+            if (!containCaseBeforeOrAfterDefault(ast)
+                    && (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
+                        || defaultGroupAST.getColumnNo()
+                            != lastGroupAST.getColumnNo())) {
                 log(ast, MSG_KEY);
             }
         }
     }
+
+    /**
+     * It returns true if after or before default there is a case or it
+     * returns false if before or after default there is no case statement.
+     *
+     * @param ast - root node.
+     * @return true if contains case before or after default or false.
+     */
+    private boolean containCaseBeforeOrAfterDefault(DetailAST ast) {
+        boolean containsCase = false;
+        if (ast.getPreviousSibling() != null) {
+            containsCase = true;
```

Comment:
if you find yourself assigning a `true` value to a variable based on the `true` result of an `if` statement, then it seems you can just drop the `if` statement and assign the variable the condition. (this sounds like it should be a check possibly)
Since the next group does the same thing, this whole method can just be a `return` statement.